currency-rates/internal/handlers/rate_test.go

```diff
@@ -51,7 +51,7 @@ func TestGetCurrentRate(main *testing.T) {
 		require.Equal(t, w.Result().StatusCode, http.StatusOK)
 	})
 
-	main.Run("failed returned rates", func(t *testing.T) {
+	main.Run("failure: returned rates", func(t *testing.T) {
```

Comment:
Use table tests or separate methods. main.Run has not benefit. You just putting all in one violating single responsibility.

Reply:
I didn't get this comment in the context of the last one, so I rewrote it to suites